src/posit/connect/external/snowflake.py

```diff
@@ -0,0 +1,44 @@
+from typing import Optional
+
+from . import _is_local
+from ..client import Client
+
+"""
+NOTE: The APIs in this module are provided as a convenience and are subject to breaking changes.
+"""
+
+class PositAuthenticator:
```

Comment:
Is this intended to work across Posit products? Would PositConnectAuthentication be more apt? Or even `Authenticator`?


Also, should we consider changing external to integrations? This seems fairly self-documenting...

```py
from posit.connect.integrations.snowflake import Authenticator

auth = Authenticator()
```

Reply:
I was following the naming style that we landed on here: https://github.com/posit-dev/posit-sdk-py/pull/252#discussion_r1705997523

> Also, should we consider changing external to integrations? This seems fairly self-documenting...

Since these APIs are still evolving I like having them under the `external` namespace for now. Maybe we consider re-naming and moving under the `integrations` namespace once we're more confident there wont be further breaking changes?

Reply:
Thanks! At least I'm consistent... 😛